src/games/strategy/util/FileUtil.java

```diff
@@ -0,0 +1,44 @@
+package games.strategy.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.File;
+
+
+/**
+ * A collection of utilities for working with files.
+ */
+public final class FileUtil {
+  private static final String PERIOD = ".";
+
+  private FileUtil() {
+    // do nothing
+  }
+
+  /**
+   * Appends the specified extension to the specified file if it is absent.
+   *
+   * <p>
+   * The file name extension comparison is case insensitive.
+   * </p>
+   *
+   * @param file The file to which the extension will be appended; must not be {@code null}.
+   * @param extension The extension to append; must not be {@code null}.
+   *
+   * @return The file, possibly with the extension appended; never {@code null}.
+   *
+   * @throws IllegalArgumentException If {@code extension} starts with a period.
+   */
+  public static File appendExtensionIfAbsent(final File file, final String extension) {
+    checkNotNull(file);
+    checkNotNull(extension);
+    checkArgument(!extension.startsWith(PERIOD), "extension must not start with a period");
```

Comment:
There is a principle to be lenient on input, but strict on output. In this case that would mean we should try to accept either type of input, with leading period or not.

We can probably overcome this by having a pretty simple intermediate variable, eg:
```
String extensionWithPeriod = extension.startsWith(PERIOD) ? extension : PERIOD + extension
```